src/ast.ts

```diff
@@ -381,16 +406,17 @@ export class Symbols {
 
     if (this[type][name]) {
       this.log.error("Duplicated " + type + " definition: " + name + ".");
-      return;
+    } else {
+      this[type][name] = object;
     }
-    this[type][name] = object;
+    return this[type][name];
   }
 
-  registerFunction(name: string, params: string[], body: Exp) {
-    this.register('functions', name, method(params, body));
+  registerFunction(name: string, params: string[], body: Exp): Method {
```

Comment:
register function is just a helper to consolidate error reporting in one place while updated the various parts of the symbol table.